server/interfaces/interfaces.go

```diff
@@ -234,6 +234,7 @@ type FileCache interface {
 
 type SchedulerService interface {
 	RegisterNode(stream scpb.Scheduler_RegisterNodeServer) error
+	RegisterExecutor(stream scpb.Scheduler_RegisterExecutorServer) error
```

Comment:
Some other ideas, just thinking out loud:

RegisterForExecution
RegisterForWork
RegisterAndStreamWork

Reply:
Renamed to RegisterAndStreamWork